src/ViewHelpers/DebugViewHelper.php

```diff
@@ -70,6 +64,83 @@ public function render() {
 			return (is_object($expressionToExamine) ? get_class($expressionToExamine) : gettype($expressionToExamine));
 		}
 
-		return var_export($expressionToExamine, TRUE);
+		$html = $this->arguments['html'];
+		$levels = $this->arguments['levels'];
+		return static::dumpVariable($expressionToExamine, $html, 1, $levels);
 	}
+
+	/**
+	 * @param mixed $variable
+	 * @param boolean $html
+	 * @param integer $level
+	 * @param integer $levels
+	 * @return string
+	 */
+	protected static function dumpVariable($variable, $html, $level, $levels) {
+		$typeLabel = is_object($variable) ? get_class($variable) : gettype($variable);
+
+		if (!$html) {
+			if (is_scalar($variable)) {
+				$string = sprintf('%s %s', $typeLabel, var_export($variable, TRUE)) . PHP_EOL;
+			} else {
+				$string = sprintf('%s: ', $typeLabel);
+				if ($level > $levels) {
+					$string .= '*Recursion limited*';
+				} else {
+					$string .= PHP_EOL;
+					foreach (static::getValuesOfNonScalarVariable($variable) as $property => $value) {
+						$string .= sprintf(
+							'%s"%s": %s',
+							str_repeat('  ', $level),
+							$property,
+							static::dumpVariable($value, $html, $level + 1, $levels)
+						);
+					}
+				}
+			}
+		} else {
+			if (is_scalar($variable)) {
+				$string = sprintf('<code>%s = %s</code>', $typeLabel, var_export($variable));
```

Comment:
Everything here that it put into HTML must be HTML encoded with htmlspecialchars!